### ISSUE TITLE
[operator] ansible kubernetes.core collection update

### ISSUE THREAD

**Issue by jmazzitelli** [kiali/kiali#7476]
The ansible collection `kubernetes.core` is used extensively in the operator. We are on a very old one, but that is because that is what ships with the ansible operator base images:

The last four base images:
```
$ for v in 4.12 4.13 4.14 4.15; do echo -n "v$v -->" && podman run --rm -it --entrypoint "" quay.io/openshift/origin-ansible-operator:$v  grep version .ansible/collections/ansible_collections/kubernetes/core/MANIFEST.json; done
v4.12 -->  "version": "2.2.3",
v4.13 -->  "version": "2.2.3",
v4.14 -->  "version": "2.4.0",
v4.15 -->  "version": "2.4.0",
```

It is up to v5.0.0 now. Currently, we are [using 2.2.3 as defined here](https://github.com/kiali/kiali-operator/blob/v1.86.0/requirements.yml#L37-L38).

We should move to at least 4.0.0 to pick up a [fix](https://github.com/ansible-collections/kubernetes.core/pull/676) to a bug that is known to affect the kiali operator: https://github.com/ansible-collections/kubernetes.core/commit/1955989278e3c3084c8a6fdebedf129a89041237

**Comment by jmazzitelli** [comment]
A quick run of the molecule tests after updating the collection to 4.0.0 resulted in 5 test failures. I don't know the cause, but I ran these tests twice and they failed both times. So it is likely a simple upgrade of the library isn't enough:

```
$ ./hack/run-molecule-tests.sh --client-exe "$(which kubectl)"  --minikube-exe $(which minikube) --minikube-profile ci --cluster-type minikube --use-dev-images true -at "accessible-namespaces-test cluster-wide-access-test grafana-test jaeger-test rolling-restart-test"
...
=====================
=== TEST RESULTS: ===
=====================

              accessible-namespaces-test... FAILURE [1m 56s]
                cluster-wide-access-test... FAILURE [2m 6s]
                            grafana-test... FAILURE [1m 50s]
                             jaeger-test... FAILURE [1m 54s]
                    rolling-restart-test... FAILURE [1m 49s]
```

**Comment by jmazzitelli** [comment]
Here's a workaround for someone that wants to update the collection on an operator that is already installed (in this example, it is installed in OpenShift's OLM infrastructure so `oc` is being used; `kubectl` would work as well):

1. Delete any Kiali CRs for now (if installed via Maistra, you need to set the kiali enabled flag to "false" in the SMCP)
2. Tell the operator to enable writes on the file system:
`oc patch deployment kiali-operator -n openshift-operators --type='json' -p='[{"op": "replace", "path": "/spec/template/spec/containers/0/securityContext/readOnlyRootFilesystem", "value": false}]'`
3. Confirm that the filesystem is no longer read-only (this should return "false"):
`oc get deployment kiali-operator -n openshift-operators -o jsonpath='{..spec.containers[0].securityContext.readOnlyRootFilesystem}'`
4. Now update the "kubernetes.core" ansible collection to at least 4.0.0 (which is when [it looks like the fix was first applied for a particular bug that affects the operator](https://github.com/ansible-collections/kubernetes.core/commit/1955989278e3c3084c8a6fdebedf129a89041237)):
`oc exec -it -n openshift-operators deployment/kiali-operator -- ansible-galaxy collection install -f kubernetes.core:=4.0.0`
5. Confirm the collection was updated (this should output "version: 4.0.0"):
`oc exec -it -n openshift-operators deployment/kiali-operator -- cat /opt/ansible/.ansible/collections/ansible_collections/kubernetes/core/MANIFEST.json | grep version`
6. Re-install your Kiali CRs (if using Maistra, just set the kiali enabled flag back to true in the SMCP).

In summary, to do the file system change and the collection upgrade, this is the command:
```
oc patch deployment kiali-operator -n openshift-operators --type='json' -p='[{"op": "replace", "path": "/spec/template/spec/containers/0/securityContext/readOnlyRootFilesystem", "value": false}]' && \
oc exec -it -n openshift-operators deployment/kiali-operator -- ansible-galaxy collection install -f kubernetes.core:=4.0.0
```

**Comment by jmazzitelli** [comment]
PRs:
* master: https://github.com/kiali/kiali-operator/pull/783
* 1.73 backport: https://github.com/kiali/kiali-operator/pull/785
* 1.86 backport: https://github.com/kiali/kiali-operator/pull/786
